reccmp/isledecomp/analysis/funcinfo.py

```diff
@@ -0,0 +1,71 @@
+"""Parsing SEH (Structured Exception Handling) data.
+https://www.openrce.org/articles/full_view/21"""
+
+import re
+import struct
+from typing import Iterator, NamedTuple
+from reccmp.isledecomp.formats import PEImage
+
+
+# Magic strings:
+# - 0x19930520: up to VC6
+# - 0x19930521: VC7.x(2002-2003)
+# - 0x19930522: VC8 (2005)
+FUNCINFO_MAGIC_RE = re.compile(rb"\x20\x05\x93\x19", flags=re.S)
+
+
+# Match `mov eax, ____` instructions followed by jmp opcode. `B8 .... E9`
+MOV_EAX_RE = re.compile(rb"(?=\xb8(.{4})\xe9)", flags=re.S)
+
+
+class FuncInfo(NamedTuple):
+    addr: int
+    unwinds: list[tuple[int, int]]
+
+
+def find_funcinfo_offsets_in_buffer(buf: bytes) -> Iterator[int]:
+    """Return offsets of the FuncInfo magic number."""
+    for match in FUNCINFO_MAGIC_RE.finditer(buf):
+        yield match.start()
+
+
+def find_funcinfo_in_buffer(buf: bytes, base_addr: int) -> Iterator[FuncInfo]:
+    """Parse the FuncInfo struct and return its location."""
+    for ofs in find_funcinfo_offsets_in_buffer(buf):
+        # TODO: The structure may vary depending on the magic string.
+        # We support format 19930520 to start.
+        (n_unwind, unwind_addr) = struct.unpack_from("<4x2I", buf, offset=ofs)
```

Comment:
```suggestion
        # max_state: number of entries in unwind table
        # p_unwind_map: table of unwind destructors
        (max_state, p_unwind_map) = struct.unpack_from("<4x2I", buf, offset=ofs)
```
I'd stick to the terminology of openRCE